### ISSUE TITLE
Editable expression and live result

### ISSUE THREAD

**Issue by regs01** [Microsoft/calculator#150]
The app needs an ability to edit an expression, i.e. place cursor anywhere and add something in between. With live result. Just like any Android calculator out there.

**Comment by chibicitiberiu** [comment]
SpeedCrunch is a desktop application which does a similar thing, it could be a good place for inspiration.

**Comment by MicrosoftIssueBot** [comment]
This is your friendly Microsoft Issue Bot. I've seen this issue come in and have gone to tell a human about it.

**Comment by grochocki** [comment]
Thanks for the feature suggestion! This sound like an interesting idea. We have a [process in place](https://github.com/Microsoft/calculator/blob/master/docs/NewFeatureProcess.md) for feature suggestions that impact the end-user experience, which includes a template that helps frame the feature pitch. Please update using that template.

**Comment by grochocki** [comment]
This appears to be a duplicate of #163, so lets continue the discussion there. Let me know if I am mistaken and this is different.

**Comment by miloush** [comment]
I filed #163 separately because my understanding was this request is for updating the result live as the expression is edited, while my proposal was just to get on par with the Win32 calc, i.e. update the result once the edit session is finished.

**Comment by regs01** [comment]
Well you can update yours. It can be merged. No matter if it's current expression or expression from history it's still should be editable.

**Comment by grochocki** [comment]
Ah, I understand now. @regs01's idea focuses on the red rectangle, while @miloush's proposal focuses on the green. Since the UI has deviated quite a bit from Win32 calc, feel free to reactivate this and treat these as two separate ideas, or if you would prefer to collaborate on #163 and expand its scope, that is fine too.

![image](https://user-images.githubusercontent.com/1414423/54011770-907b2e80-4128-11e9-8b89-c45fa004548b.png)
